### ISSUE TITLE
build: configure fails to process libnatpmp from depends for HOST=x86_64-w64-mingw32

### ISSUE THREAD

**Issue by hebasto** [bitcoin/bitcoin#21350]
From the `config.log`:
```
configure:25743: checking for initnatpmp in -lnatpmp
configure:25768: x86_64-w64-mingw32-g++ -std=c++17 -o conftest.exe -pipe -O2  -fno-extended-identifiers -I/home/hebasto/GitHub/gui/depends/x86_64-w64-mingw32/include/  -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS -D_MT -DWIN32 -D_WINDOWS -D_WIN32_WINNT=0x0601 -D_WIN32_IE=0x0501 -DWIN32_LEAN_AND_MEAN -D_FILE_OFFSET_BITS=64 -DPROVIDE_FUZZ_MAIN_FUNCTION -L/home/hebasto/GitHub/gui/depends/x86_64-w64-mingw32/lib  -Wl,--major-subsystem-version -Wl,6 -Wl,--minor-subsystem-version -Wl,1 conftest.cpp -lnatpmp  -lssp -liphlpapi -lshlwapi -lws2_32 -ladvapi32 -luuid -loleaut32 -lole32 -lcomctl32 -lshell32 -lwinmm -lcomdlg32 -lgdi32 -luser32 -lkernel32  >&5
/usr/bin/x86_64-w64-mingw32-ld: /home/hebasto/GitHub/gui/depends/x86_64-w64-mingw32/lib/libnatpmp.a(natpmp.o):natpmp.c:(.text+0x16f): undefined reference to `natpmp_gettimeofday'
/usr/bin/x86_64-w64-mingw32-ld: /home/hebasto/GitHub/gui/depends/x86_64-w64-mingw32/lib/libnatpmp.a(natpmp.o):natpmp.c:(.text+0x1c8): undefined reference to `natpmp_gettimeofday'
/usr/bin/x86_64-w64-mingw32-ld: /home/hebasto/GitHub/gui/depends/x86_64-w64-mingw32/lib/libnatpmp.a(natpmp.o):natpmp.c:(.text+0x3d0): undefined reference to `__imp_readnatpmpresponse'
/usr/bin/x86_64-w64-mingw32-ld: /home/hebasto/GitHub/gui/depends/x86_64-w64-mingw32/lib/libnatpmp.a(natpmp.o):natpmp.c:(.text+0x3f0): undefined reference to `natpmp_gettimeofday'
collect2: error: ld returned 1 exit status
configure:25768: $? = 1
configure: failed program was:
| /* confdefs.h */
| #define PACKAGE_NAME "Bitcoin Core"
| #define PACKAGE_TARNAME "bitcoin"
| #define PACKAGE_VERSION "21.99.0"
| #define PACKAGE_STRING "Bitcoin Core 21.99.0"
| #define PACKAGE_BUGREPORT "https://github.com/bitcoin/bitcoin/issues"
| #define PACKAGE_URL "https://bitcoincore.org/"
| #define HAVE_CXX17 1
| #define STDC_HEADERS 1
| #define HAVE_SYS_TYPES_H 1
| #define HAVE_SYS_STAT_H 1
| #define HAVE_STDLIB_H 1
| #define HAVE_STRING_H 1
| #define HAVE_MEMORY_H 1
| #define HAVE_STRINGS_H 1
| #define HAVE_INTTYPES_H 1
| #define HAVE_STDINT_H 1
| #define HAVE_UNISTD_H 1
| #define LT_OBJDIR ".libs/"
| #define USE_ASM 1
| #define ENABLE_SSE41 1
| #define ENABLE_AVX2 1
| #define ENABLE_SHANI 1
| #define HAVE_LIBKERNEL32 1
| #define HAVE_LIBUSER32 1
| #define HAVE_LIBGDI32 1
| #define HAVE_LIBCOMDLG32 1
| #define HAVE_LIBWINMM 1
| #define HAVE_LIBSHELL32 1
| #define HAVE_LIBCOMCTL32 1
| #define HAVE_LIBOLE32 1
| #define HAVE_LIBOLEAUT32 1
| #define HAVE_LIBUUID 1
| #define HAVE_LIBADVAPI32 1
| #define HAVE_LIBWS2_32 1
| #define HAVE_LIBSHLWAPI 1
| #define HAVE_LIBIPHLPAPI 1
| #define HAVE___INT128 1
| #define HAVE_PTHREAD_PRIO_INHERIT 1
| #define HAVE_PTHREAD 1
| #define _FILE_OFFSET_BITS 64
| #define HAVE_DECL_STRERROR_R 0
| #define HAVE_LIBSSP 1
| #define HAVE_STDIO_H 1
| #define HAVE_STDLIB_H 1
| #define HAVE_UNISTD_H 1
| #define HAVE_STRINGS_H 1
| #define HAVE_SYS_TYPES_H 1
| #define HAVE_SYS_STAT_H 1
| #define HAVE_DECL_GETIFADDRS 0
| #define HAVE_DECL_FREEIFADDRS 0
| #define HAVE_DECL_STRNLEN 1
| #define HAVE_DECL_DAEMON 0
| #define HAVE_DECL_PIPE2 0
| #define HAVE_DECL_LE16TOH 0
| #define HAVE_DECL_LE32TOH 0
| #define HAVE_DECL_LE64TOH 0
| #define HAVE_DECL_HTOLE16 0
| #define HAVE_DECL_HTOLE32 0
| #define HAVE_DECL_HTOLE64 0
| #define HAVE_DECL_BE16TOH 0
| #define HAVE_DECL_BE32TOH 0
| #define HAVE_DECL_BE64TOH 0
| #define HAVE_DECL_HTOBE16 0
| #define HAVE_DECL_HTOBE32 0
| #define HAVE_DECL_HTOBE64 0
| #define HAVE_DECL_BSWAP_16 0
| #define HAVE_DECL_BSWAP_32 0
| #define HAVE_DECL_BSWAP_64 0
| #define HAVE_BUILTIN_CLZL 1
| #define HAVE_BUILTIN_CLZLL 1
| #define HAVE_DEFAULT_VISIBILITY_ATTRIBUTE 1
| #define HAVE_DLLEXPORT_ATTRIBUTE 1
| #define HAVE_FDATASYNC 0
| #define HAVE_O_CLOEXEC 0
| #define HAVE_STD__SYSTEM 1
| #define HAVE_SYSTEM HAVE_STD__SYSTEM || HAVE_WSYSTEM
| #define QT_STATICPLUGIN 1
| #define QT_QPA_PLATFORM_MINIMAL 1
| #define QT_QPA_PLATFORM_WINDOWS 1
| #define USE_BDB 1
| #define USE_SQLITE 1
| #define HAVE_MINIUPNPC_MINIWGET_H 1
| #define HAVE_MINIUPNPC_MINIUPNPC_H 1
| #define HAVE_MINIUPNPC_UPNPCOMMANDS_H 1
| #define HAVE_MINIUPNPC_UPNPERRORS_H 1
| #define HAVE_NATPMP_H 1
| /* end confdefs.h.  */
| 
| /* Override any GCC internal prototype to avoid an error.
|    Use char because int might match the return type of a GCC
|    builtin and then its argument prototype would still apply.  */
| #ifdef __cplusplus
| extern "C"
| #endif
| char initnatpmp ();
| int
| main ()
| {
| return initnatpmp ();
|   ;
|   return 0;
| }
configure:25777: result: no
```

**Comment by fanquake** [comment]
I'd assume fixed by #21209.

**Comment by hebasto** [comment]
@fanquake 
> I'd assume fixed by #21209.

Nope. Just tested.
